BullsAndCows/BullsAndCowsGame.cs

```diff
@@ -1,20 +1,89 @@
 ﻿using System;
+using System.Linq;
 
 namespace BullsAndCows
 {
     public class BullsAndCowsGame
     {
         private readonly SecretGenerator secretGenerator;
+        private readonly string secret;
+        private int gameCounter = 1;
+
         public BullsAndCowsGame(SecretGenerator secretGenerator)
         {
+            this.CanContinue = true;
             this.secretGenerator = secretGenerator;
+            this.secret = this.secretGenerator.GenerateSecret();
+            Console.WriteLine(this.secret);
         }
 
-        public bool CanContinue => true;
+        public bool CanContinue { get; set; }
 
         public string Guess(string guess)
         {
-            throw new NotImplementedException();
+            if (gameCounter >= 6)
+            {
+                CanContinue = false;
+            }
+
+            while (!IsGuessVlid(guess))
+            {
+                Console.WriteLine("input is not valid, please input 4 different digital, use space to separate each other");
+                guess = Console.ReadLine();
+            }
+
+            var guessWithoutSpace = guess.Replace(" ", string.Empty);
+            return this.Compare(this.secret, guessWithoutSpace);
+        }
+
+        private bool IsGuessVlid(string guess)
+        {
+            var guessWithoutSpace = guess.Replace(" ", string.Empty);
+            bool moreThanOnce = guessWithoutSpace.GroupBy(x => x).Any(g => g.Count() > 1);
+            if (guessWithoutSpace.Length == 4 && !moreThanOnce)
+            {
+                return true;
+            }
+
+            return false;
```

Comment:
Could shorten the code. This can be directly replaced by return guessWithoutSpace.Length == 4 && !moreThanOnce